sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/LimitPushdownThroughWindowSuite.scala

```diff
@@ -155,36 +155,164 @@ class LimitPushdownThroughWindowSuite extends PlanTest {
       WithoutOptimize.execute(correctAnswer.analyze))
   }
 
-  test("Should not push down if partitionSpec is not empty") {
+  test("Should not push down when child's maxRows smaller than limit value") {
```

Comment:
Can we do not move the order? To reduce the change.

Reply:
> Can we do not move the order? To reduce the change.

Done